src/v1/sources/shopify/pixelTransform.js

```diff
@@ -20,6 +29,43 @@ const NO_OPERATION_SUCCESS = {
   },
   statusCode: 200,
 };
+const extractCartToken = (cartToken) => {
+  if (typeof cartToken !== 'string') {
+    logger.error(`Cart token is not a string`);
+    return undefined;
+  }
+  const cartTokenParts = cartToken.split('/');
+  return cartTokenParts[3];
+};
+
+const handleCartTokenRedisOperations = async (inputEvent, clientId) => {
+  try {
+    const cartTokenLocation = pixelEventToCartTokenLocationMapping[inputEvent.name];
+    if (!cartTokenLocation) {
+      stats.increment('shopify_pixel_cart_token_not_found', {
+        event: inputEvent.name,
+        writeKey: inputEvent.query_parameters.writeKey,
+      });
+      return;
+    }
+    const unparsedCartToken = _.get(inputEvent, cartTokenLocation);
+    const cartToken = extractCartToken(unparsedCartToken);
+    inputEvent.anonymousId = clientId;
```

Comment:
Assigning anonymousId is common for all client-side events, not only for the ones where cartToken is present

Reply:
Addressed it

Reply:
I still see in the latest change `inputEvent.anonymousId = clientId;`

Reply:
updated